nomad/fsm.go

```diff
@@ -625,6 +625,49 @@ func (n *nomadFSM) applyUpsertJob(msgType structs.MessageType, buf []byte, index
 		panic(fmt.Errorf("failed to decode request: %v", err))
 	}
 
+	// Preserve the existing task group counts and resources, if requested. Do
+	// this here to avoid race conditions when registering and scaling jobs.
+	if req.PreserveCounts || req.PreserveResources {
+		existingJobID := req.Job.ID
+		if existingJobID != "" {
+			existingJob, err := n.state.JobByID(nil, req.Namespace, existingJobID)
+			if err != nil {
+				n.logger.Error("JobByID lookup for existing job failed", "existing_job_id", existingJobID, "namespace", req.Namespace, "error", err)
```

Comment:
The caller ends up logging any errors, so we can skip this. As it turns out it's also all but impossible to return an error here, because the errors we get from go-memdb here are _always_ programmer errors (ex. wrong index) rather than runtime errors.